library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -1064,6 +1061,77 @@ protected int parseAudioChannelConfiguration(XmlPullParser xpp)
     return audioChannels;
   }
 
+  // Selection flag parsing.
+
+  protected int parseSelectionFlags(Descriptor roleDescriptor) {
+    return "urn:mpeg:dash:role:2011".equals(roleDescriptor.schemeIdUri)
+      && "main".equals(roleDescriptor.value) ? C.SELECTION_FLAG_DEFAULT : 0;
+  }
+
+  // Role and Accessibility parsing.
+
+  protected @C.Role int parseRole(List<Descriptor> roleDescriptors) {
+    for (int i = 0; i < roleDescriptors.size(); i++) {
+      Descriptor descriptor = roleDescriptors.get(i);
+      if ("urn:mpeg:dash:role:2011".equalsIgnoreCase(descriptor.schemeIdUri) && descriptor.value != null) {
+        switch (descriptor.value) {
+          case "main":
+            return C.ROLE_MAIN;
+          case "alternate":
+            return C.ROLE_ALTERNATE;
+          case "supplementary":
+            return C.ROLE_SUPPLEMENTARY;
+          case "commentary":
+            return C.ROLE_COMMENTARY;
+          case "dub":
+            return C.ROLE_DUB;
+          case "emergency":
+            return C.ROLE_EMERGENCY;
+          case "caption":
+            return C.ROLE_CAPTION;
+          case "sign":
+            return C.ROLE_SIGN;
+          default:
+            return C.ROLE_UNSET;
+        }
+      }
+    }
+    return C.ROLE_UNSET;
+  }
+
+  protected @C.Accessibility int parseAccessibility(List<Descriptor> accessibilityDescriptors) {
```

Comment:
If going with my suggested approach, this method will return C.RoleFlags, which is slightly strange. But then it's somewhat strange DASH-IF separated use of `urn:mpeg:dash:role:2011` across `Role` and `Accessibility` elements to start with, in my opinion :)... They should then be OR'd with the `parseRole` result to get the final flags.

You could probably have a single method parse "urn:mpeg:dash:role:2011" that you call from both `parseRole` and `parseAccessibility`. I think we'd probably want to handle things like "description" appearing as Role elements as well.

Reply:
> You could probably have a single method parse "urn:mpeg💨role:2011" that you call from both parseRole and parseAccessibility. I think we'd probably want to handle things like "description" appearing as Role elements as well.

This would mean that the roleFlags would have a common flag for role's "description" and accessibility's "description" values. In this case we couldn't decide if the "description" came from role or from accessibility. 

Even if a couple of values could be the same for role and accessibility, I would store them as separate flags, as at the end we need to know where a value comes from (role or accessibility descriptor).

Reply:
Nevermind, probably we can merge these as you said.